src/main/scala/chisel3/util/Cat.scala

```diff
@@ -19,7 +20,7 @@ object Cat {
   /** Concatenates the argument data elements, in argument order, together. The first argument
     * forms the most significant bits, while the last argument forms the least significant bits.
     */
-  def apply[T <: Bits](a: T, r: T*): UInt = apply(a :: r.toList)
+  def apply[T <: Bits](a: T, r: T*)(implicit sourceInfo: SourceInfo): UInt = apply(a :: r.toList)
```

Comment:
Actually, we need to use `do_apply` macros here because one could do a bit extraction immediately after a `Cat`.

Reply:
I do not know what that means, can you point to an example?

Reply:
It's used a lot in `Bits.scala`: https://github.com/chipsalliance/chisel3/blob/44165a259bb16733a41798edca6b554b13f1d54a/core/src/main/scala/chisel3/Bits.scala#L73.

Here's a link to me explaining what this is and how it works in the Chisel internals talk at the CCC last year: https://youtu.be/2_WPFk8lriE?t=1166

Reply:
ok, i believe I have this right now.